cmake/traccc-acts.cmake

```diff
@@ -0,0 +1,31 @@
+# TRACCC library, part of the ACTS project (R&D line)
+#
+# (c) 2021 CERN for the benefit of the ACTS project
+#
+# Mozilla Public License Version 2.0
+
+# Guard against multiple includes.
+include_guard(GLOBAL)
+
+# Try to find ACTS installed on the system.
+find_package(Acts QUIET)
+
+# If it was found, then we're finished.
+if(Acts_FOUND)
+   # Call find_package again, just to nicely print where it is picked up from.
+   find_package(Acts)
+   return()
+endif()
+
+# Tell the user what's happening.
+message(STATUS "Building ACTS as part of the traccc project")
+
+# Declare where to get ACTS from.
+FetchContent_Declare(
+   Acts
+   GIT_REPOSITORY "https://github.com/acts-project/acts.git"
+   GIT_TAG "v9.2.0"
+)
+
+# Make the ACTS source available to the rest of the build system.
+FetchContent_MakeAvailable(Acts)
```

Comment:
:thinking: I wonder if we may want to use the same formalism that I used for VecMem.

```cmake
FetchContent_Populate( VecMem )
add_subdirectory( "${vecmem_SOURCE_DIR}" "${vecmem_BINARY_DIR}"
   EXCLUDE_FROM_ALL )
```

The advantage of that is that only those parts of VecMem are built that are explicitly needed by traccc. So in VecMem's case the unit tests are not built by default.

I know that Acts works a little differently. It doesn't build unit tests by default, you have to explicitly turn them on. Still, I liked this setup quite a bit, where the "included project" would not add any targets to `ALL` itself. It would just provide the targets for components in this main project to use.

However I don't feel super strongly about it in the end...

Reply:
So if I understand what you're saying, using the `FetchContent_MakeAvailable` formalism adds every target defined by vecmem (which is not explicitly disabled by a CMake flag) to `ALL`, and as such running `make` without any specifiers would do extra work building those unnecessary parts, while your approach does not implicitly add them to `ALL` and, as a result, parts of the code base are only built if they are actually required by one of the targets within traccc as a dependency? If that is the case, I agree it makes sense to do it your way.

Reply:
Yes, empirical proof (me testing it on my laptop) shows significantly faster builds when Acts is never actually used as a dependency anywhere in the code. It's unlikely that this would actually ever happen in practice, but I agree it is a better solution.

That said it seems like very strange design that `FetchContent_MakeAvailable` does not have a `EXCLUDE_FROM_ALL` option. Perhaps we should open a merge request on CMake...

Reply:
Doesn't `FetchContent_MakeAvailable` literally do the same thing as @krasznaa wrote above? If so, we could just use manual `Populate` and `add_subdirectory` with `EXCLUDE_FROM_ALL`.

Reply:
_Almost_, but not quite. The `FetchContent_MakeAvailable` macro is defined here: https://github.com/Kitware/CMake/blob/master/Modules/FetchContent.cmake#L1183, and the subdirectory is added in line 1211. It does not specify the `EXCLUDE_FROM_ALL` flag, and I really don't understand why. Anyhow, I've updated the pull request to use the formalism suggested by Attila.